README.md

```diff
@@ -64,20 +64,51 @@ Contract code size can be benched by running:
 yarn bench:code-size
 ```
 
-### Deploying Contracts
+### Deploying Contracts: Proposal-Transactions Creation
+
+The contracts are deployed by the Gnosis DAO using the Zodiac module. 
+In the following, it is show on to build the tx proposed to the Gnosis DAO with a script.
+
+The deployment happens on two chains: Ethereum-Chain and Gnosis-Chain. 
+At first, a deployment helper contract - called BridgedTokenDeployer - is deployed on Gnosis-Chain. 
+This BridgedTokenDeployer contains the information to run the CowProtocolVirtualToken deployment on Gnosis-Chain. 
+This contract will later be triggered from the Ethereum-Chain via the Omni-Bridge. 
+The main part of the deployment is done on Ethereum-Chain. 
+The GnosisDAO will initiate all necessary transactions to create the different safes, create the CowProtocolToken and CowProtocolVirtualToken. 
+Furthermore, the GnosisDAO will bridge one CowProtocolToken to the Omni-Bridge in order to trigger the bridge to deploy the bridged CowProtocolToken also on Gnosis-Chain. 
+Last, but not least, the GnosisDao will deploy over the bridge also a new Community Safe on Gnosis Chain and trigger the CowProtocolVirtualToken deployment on the BridgedTokenDeployer over the Omni-Bridge.
+
+The deployment has the following inputs:
+- .env file for Ethereum-Chain. See [example](.env.sample)
```

Comment:
I'd put this at the end, and mention that it should be sourced in the command line before executing the script.